lib/cog_api/resources/role.ex

```diff
@@ -4,5 +4,6 @@ defmodule CogApi.Resources.Role do
   defstruct [
     :id,
     :name,
+    :permissions,
```

Comment:
Can we test that this is in fact taking in permissions in the `http` test? Also that it is parsing them as `%Permission{}` structs?

You'll likely have to change this to: `permissions: [%CogApi.Resources.Permission{}]` for the parsing to work.